### ISSUE TITLE
Add some vim-like keybindings

### ISSUE THREAD

**Issue by istudyatuni** [faldor20/jj_tui#8]
Could you please add a `j/k` keybindings for scrolling log/files/bookmarks down/up by 1 line

And probably also `Shift+j/k` for scrolling by half of screen (height of window)

**Comment by faldor20** [comment]
Yeah, sure. I think I'll probably just add in key mapping, the system is designed entirely around supporting key remapping, so It shouldn't be too hard to implement, I don't have a whole lot of time, but give it a moth or so and this should be solved :)

**Comment by MrDwarf7** [comment]
Doesn't seem like up/down -> j/k is exposed - would love this too.

**Comment by faldor20** [comment]
Yeah, I know, it's quite a bit more annoying to support those remapping sadly.

Next time I sit down and do some work on it, I'll see about sorting that out.


In the meantime, I can highly recommend the Kanata keyabord remapping software, I have my home row setup to produce arrow keys in a layer activated by holding caps lock and it's useful in heaps of applications, but I use colemak so hjk doesn't work for me anyway 😅

**Comment by MrDwarf7** [comment]
@faldor20 I use both a ZSA keyboard and a ZMK corne, definitely know how good remaps are lol. Not sure how I'd manage without them between home->work computers tbh.
I stumbled upon this via the JJ docs and have packaged it for the AUR (Yet to go through release for it though).

I'd give implementing it a go but I literally only installed OCaml like a week ago and have yet to look through how to write a single thing in the language lol, only barely know how to read it by inference from other languages (Rust, Nix and poking around in Gleam/Elixir).

If I happen to continue down the OCaml learning route I may be able to attempt a PR of some sort in the future if it's before this gets attended to/implemented (Though expect the code to be very newbie if that's the case lol).

**Comment by faldor20** [comment]
Hey folks, this is now complete.  Get the latest version of jj_tui and you are free to remap the arrow keys however you please, vim keys are mapped by default.
